asmcomp/power/emit.mlp

```diff
@@ -177,6 +177,28 @@ let emit_tocload emit_dest dest entry =
 
 (* Output a load or store operation *)
 
+let load_mnemonic = function
+  | Byte_unsigned -> "lbz"
+  | Byte_signed -> "lbz"
+  | Sixteen_unsigned -> "lhz"
+  | Sixteen_signed -> "lha"
+  | Thirtytwo_unsigned -> "lwz"
+  | Thirtytwo_signed -> "lwa"
+  | Word_int | Word_val -> "ld"
+  | Single -> "lfs"
+  | Double -> "lfd"
+
+let store_mnemonic = function
+  | Byte_unsigned | Byte_signed -> "stb"
+  | Sixteen_unsigned | Sixteen_signed -> "sth"
+  | Thirtytwo_unsigned | Thirtytwo_signed -> "stw"
+  | Word_int | Word_val -> "std"
+  | Single -> "stfs"
```

Comment:
This was `assert false` in the original code. Does this mean that using float32 bigarrays could cause compilation errors on power, and this PR fixes that ? Or is there some code somewhere that makes sure we don't generate float32 stores ?

Reply:
No, the emitter has a special case for `Lop(Istore(Single, addr, _assignment))` (lines 729-732 in the unmodified source code), so this was an impossible code path until moved to an helper function.